plugins/javaembed/javaembed.cpp

```diff
@@ -120,12 +124,58 @@ class CheckedJNIEnv : private JNIEnv
         }
         return ret;
     }
+
+    void ignoreException()
+    {
+        if (JNIEnv::ExceptionCheck())
+        {
+            DBGLOG("javaembed: Ignoring unexpected java exception while processing exception");
+            JNIEnv::ExceptionDescribe();
+            JNIEnv::ExceptionClear();
+        }
+    }
+
+    void traceException(jthrowable exception)
+    {
+        // Don't use auto-checking in here, as we are already inside exception-checking code
+        jstring msg = (jstring) JNIEnv::CallObjectMethod(exception, throwable_toString);
+        ignoreException();
+        const char *text = JNIEnv::GetStringUTFChars(msg, 0);
+        DBGLOG("javaembed: exception: %s", text);
+        JNIEnv::ReleaseStringUTFChars(msg, text);
+        ignoreException();
+        JNIEnv::DeleteLocalRef(msg);
+
+        jobjectArray frames = (jobjectArray) JNIEnv::CallObjectMethod(exception, throwable_getStackTrace);
+        ignoreException();
+        jsize length = JNIEnv::GetArrayLength(frames);
```

Comment:
are we guaranteed frames will be valid if we ignored a possible exception? It might not matter if JNIEnv::GetArrayLength sets length<= 0

Reply:
Unless there's a logic error, getStackTrace should always succeed. the main reason for the "ignoreException" is that there is a mode of running JNI code that detects whether you have checked the exception flag after each call. I could throw an internal error inside ignoreException if it detects an error.